mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -169,6 +169,41 @@ private[spark] object BLAS extends Serializable with Logging {
   }
 
   /**
+   * Adds alpha * x * x.t to a matrix in-place. This is the same as BLAS's DSPR.
+   *
+   * @param U the upper triangular part of the matrix packed in an array (column major)
+   */
+  def dspr(alpha: Double, v: Vector, U: Array[Double]): Unit = {
```

Comment:
* Should be `spr` instead of `dspr`. We only use double precision.
* It might be useful to make `U` a `DenseVector` (or add a convenient method) for API consistency.
* Do you mind adding tests for both dense and sparse input?